packages/compass-aggregations/src/modules/pipeline-builder/text-editor-pipeline.ts

```diff
@@ -84,15 +79,33 @@ const reducer: Reducer<TextEditorState> = (state = INITIAL_STATE, action) => {
     action.type === CONFIRM_NEW ||
     action.type === NEW_PIPELINE
   ) {
-    const stageOperators = action.pipeline
-      ? action.pipeline.map(getStageOperator).filter(Boolean) as string[]
-      : state.stageOperators;
+    // On editor switch or reset, reset the parsed pipeline completely
+    const pipeline = action.pipeline ?? [];
```

Comment:
@mabaasit if we are completely resetting the state here, seems like a more logical thing to do here to me to not fallback to the old state, even if we failed to parse. Makes sense?

The case where pipeline is not parseable when user is editing is handled separately now

Reply:
Sounds good :)